ui/src/main/kotlin/org/jetbrains/jewel/ui/component/SplitLayout.kt

```diff
@@ -15,184 +18,407 @@ import androidx.compose.runtime.mutableStateOf
+    state: SplitLayoutState = rememberSplitLayoutState(),
+) {
+    SplitLayoutImpl(
+        first = first,
+        second = second,
+        modifier = modifier,
+        dividerStyle = dividerStyle,
+        draggableWidth = draggableWidth,
+        minFirstPaneSize = firstPaneMinWidth,
+        minSecondPaneSize = secondPaneMinWidth,
+        strategy = verticalTwoPaneStrategy(),
+        state = state,
+    )
+}
+
+public class SplitLayoutState(initialSplitFraction: Float) {
+    public var dividerPosition: Float by mutableStateOf(initialSplitFraction.coerceIn(0f, 1f))
+    public var layoutCoordinates: LayoutCoordinates? by mutableStateOf(null)
+}
+
+@Composable
+public fun rememberSplitLayoutState(initialSplitFraction: Float = 0.5f): SplitLayoutState = remember {
+    SplitLayoutState(initialSplitFraction)
+}
+
+@Composable
+private fun SplitLayoutImpl(
+    first: @Composable () -> Unit,
+    second: @Composable () -> Unit,
+    strategy: SplitLayoutStrategy,
+    modifier: Modifier,
+    draggableWidth: Dp,
+    minFirstPaneSize: Dp,
+    minSecondPaneSize: Dp,
+    dividerStyle: DividerStyle,
+    state: SplitLayoutState,
 ) {
     val density = LocalDensity.current
-    var dividerY by remember { mutableStateOf(with(density) { initialDividerPosition.roundToPx() }) }
+    var dragStartPosition by remember { mutableStateOf(0f) }
+    var currentDragPosition by remember { mutableStateOf(0f) }
+    var isDragging by remember { mutableStateOf(false) }
+    val resizeCursor = if (strategy.isHorizontal()) Cursor(Cursor.E_RESIZE_CURSOR) else Cursor(Cursor.N_RESIZE_CURSOR)
+    val defaultCursor = Cursor(Cursor.DEFAULT_CURSOR)
 
     Layout(
-        modifier = modifier,
+        modifier =
+            modifier
+                .onGloballyPositioned { coordinates -> state.layoutCoordinates = coordinates }
+                .pointerHoverIcon(PointerIcon(if (isDragging) resizeCursor else defaultCursor)),
         content = {
+            Box(Modifier.layoutId("first")) { first() }
+            Box(Modifier.layoutId("second")) { second() }
+
             val dividerInteractionSource = remember { MutableInteractionSource() }
-            first(Modifier.layoutId("first"))
+            val dividerOrientation = if (strategy.isHorizontal()) Vertical else Horizontal
+            val fillMaxDirection = if (strategy.isHorizontal()) Modifier.fillMaxHeight() else Modifier.fillMaxWidth()
+            val orientation = if (strategy.isHorizontal()) Orientation.Horizontal else Orientation.Vertical
 
             Divider(
-                orientation = Orientation.Horizontal,
-                modifier = Modifier.fillMaxHeight().layoutId("divider"),
-                color = dividerColor,
-                thickness = dividerThickness,
-                startIndent = dividerIndent,
+                orientation = dividerOrientation,
+                modifier = Modifier.then(fillMaxDirection).layoutId("divider").focusable(false),
```

Comment:
You don't need `Modifier.then()`, just start the chain with the first modifier:

```suggestion
                modifier = fillMaxDirection.layoutId("divider").focusable(false),
```